src/viam/sdk/tests/test_resource.cpp

```diff
@@ -0,0 +1,250 @@
+    vec3.set_z(9);
+    *box.mutable_dims_mm() = vec3;
+    *g.mutable_label() = "label";
+    *g.mutable_center() = pose;
+    *g.mutable_box() = box;
+
+    viam::app::v1::Orientation o;
+    viam::app::v1::Orientation_AxisAngles aa;
+    aa.set_x(10);
+    aa.set_y(11);
+    aa.set_z(12);
+    aa.set_theta(13);
+    *o.mutable_axis_angles() = aa;
+
+    viam::app::v1::Translation t;
+    t.set_x(14);
+    t.set_y(15);
+    t.set_z(16);
+
+    *frame.mutable_parent() = "parent";
+    *frame.mutable_geometry() = g;
+    *frame.mutable_orientation() = o;
+    *frame.mutable_translation() = t;
+
+    LinkConfig lc = LinkConfig::from_proto(frame);
+    BOOST_CHECK_EQUAL(lc.get_parent(), "parent");
+    BOOST_CHECK_EQUAL(lc.get_translation().x, t.x());
+    BOOST_CHECK_EQUAL(lc.get_translation().y, t.y());
+    BOOST_CHECK_EQUAL(lc.get_translation().z, t.z());
+    GeometryConfig gcfg = lc.get_geometry_config();
+    BOOST_CHECK_EQUAL(gcfg.get_label(), "label");
+    BOOST_CHECK_EQUAL(gcfg.get_pose(), pose::from_proto(pose));
+    BOOST_CHECK_EQUAL(gcfg.get_geometry_type(), GeometryType::box);
+    const auto gs = gcfg.box_proto();
+    BOOST_CHECK_EQUAL(gs.dims_mm().x(), box.dims_mm().x());
+    BOOST_CHECK_EQUAL(gs.dims_mm().y(), box.dims_mm().y());
+    BOOST_CHECK_EQUAL(gs.dims_mm().z(), box.dims_mm().z());
+
+    viam::app::v1::Frame proto_lc = lc.to_proto();
+    BOOST_CHECK_EQUAL(proto_lc.parent(), "parent");
+    BOOST_CHECK_EQUAL(proto_lc.translation().x(), t.x());
+    BOOST_CHECK_EQUAL(proto_lc.translation().y(), t.y());
+    BOOST_CHECK_EQUAL(proto_lc.translation().z(), t.z());
+    BOOST_CHECK(proto_lc.has_orientation());
+    viam::app::v1::Orientation proto_ocfg = proto_lc.orientation();
+    BOOST_CHECK_EQUAL(proto_ocfg.type_case(), viam::app::v1::Orientation::TypeCase::kAxisAngles);
+    BOOST_CHECK_EQUAL(proto_ocfg.axis_angles().x(), o.axis_angles().x());
+    BOOST_CHECK_EQUAL(proto_ocfg.axis_angles().y(), o.axis_angles().y());
+    BOOST_CHECK_EQUAL(proto_ocfg.axis_angles().z(), o.axis_angles().z());
+    BOOST_CHECK_EQUAL(proto_ocfg.axis_angles().theta(), o.axis_angles().theta());
+    BOOST_CHECK(proto_lc.has_geometry());
+    viam::common::v1::Geometry proto_gcfg = proto_lc.geometry();
+    BOOST_CHECK_EQUAL(proto_gcfg.label(), "label");
+    BOOST_CHECK_EQUAL(proto_gcfg.center().x(), pose.x());
+    BOOST_CHECK_EQUAL(proto_gcfg.center().y(), pose.y());
+    BOOST_CHECK_EQUAL(proto_gcfg.center().z(), pose.z());
+    BOOST_CHECK_EQUAL(proto_gcfg.geometry_type_case(),
+                      viam::common::v1::Geometry::GeometryTypeCase::kBox);
+    BOOST_CHECK_EQUAL(proto_gcfg.box().dims_mm().x(), box.dims_mm().x());
+    BOOST_CHECK_EQUAL(proto_gcfg.box().dims_mm().y(), box.dims_mm().y());
+    BOOST_CHECK_EQUAL(proto_gcfg.box().dims_mm().z(), box.dims_mm().z());
+}
+
+BOOST_AUTO_TEST_CASE(test_resource) {
+    ResourceConfig resource1("type");
+    BOOST_CHECK_EQUAL(resource1.api().to_string(), "rdk:type:");
```

Comment:
Out of scope for this PR, but looking at this it's pretty weird. The trailing colon, the fact that this is what we initialize for a resource (why is it the `type` that we're asking for? why not the more specific `subtype`?). From some digging I think there's definitely space to improve on the `ResourceConfig` type in general. Ticket here: https://viam.atlassian.net/browse/RSDK-6403